dotNETStandard/HealthVault.Mobile/cs/Xamarin.Core/ViewModels/MedicationEditViewModel.cs

```diff
@@ -6,38 +6,46 @@
 using System.Windows.Input;
 using HealthVault.Sample.Xamarin.Core.Services;
 using Microsoft.HealthVault.Clients;
+using Microsoft.HealthVault.Connection;
 using Microsoft.HealthVault.ItemTypes;
+using Microsoft.HealthVault.Person;
 using Microsoft.HealthVault.Vocabulary;
 using Xamarin.Forms;
 
 namespace HealthVault.Sample.Xamarin.Core.ViewModels
 {
     public class MedicationEditViewModel : ViewModel
     {
+        private readonly Medication medication;
+        private readonly IHealthVaultConnection connection;
+
         public MedicationEditViewModel(
             Medication medication,
-            IList<VocabularyItem> ingredientChoices,
-            IThingClient thingClient,
-            Guid recordId, 
+            IHealthVaultConnection connection,
```

Comment:
Also if you are doing manual injection, if you make it the responsibility of the parent class to get all of your stuff ready for you, you are increasing coupling. It was really bad in the case of Medication because the menu was preparing data for a ViewModel 3 layers deeper, passing it through every viewmodel.

Also for the samples I thought it important to keep the code that's preparing the data on the actual ViewModel that needs it, so the user doesn't need to hunt to find the code they are interested in.

---
In reply to: [117636527](https://github.com/Microsoft/healthvault-samples/pull/11#discussion_r117636527) [](ancestors = 117636527,117591813)